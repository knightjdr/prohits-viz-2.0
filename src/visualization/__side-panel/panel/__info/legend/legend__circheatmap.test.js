import { shallow } from 'enzyme';

import colorGradient from '../../../../color/color-gradient';
import CircHeatmapLegend from './legend__circheatmap';

jest.mock('../../../../color/color-gradient');
colorGradient.mockReturnValue(['#ffffff', '#0000ff', '#000000']);

const segments = [
  { name: 'attribute 1' },
  { name: 'attribute 1' },
  { name: 'attribute 1' },
];
const settings = [
  {
    abundanceCap: 50,
    color: 'blueBlack',
    minAbundance: 0,
  },
  {
    abundanceCap: 50,
    color: 'redBlack',
    minAbundance: 0,
  },
  {
    abundanceCap: 50,
    color: 'yellowBlack',
    minAbundance: 0,
  },
];

describe('CircHeatmap legend', () => {
  describe('with known line segment', () => {
    let wrapper;

    beforeAll(() => {
      colorGradient.mockClear();
      wrapper = shallow(
        CircHeatmapLegend({
          known: true,
          circHeatmapSettings: settings,
          segments,
        }),
      );
    });

    it('should to match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should render 4 g elements, 3 for gradients and one for known segment', () => {
      expect(wrapper.find('svg > g').length).toBe(4);
    });

    it('should set color gradient for outer circle', () => {
      expect(colorGradient).toHaveBeenCalledWith('blueBlack', 101, false);
    });

    it('should set color gradient for middle circle', () => {
      expect(colorGradient).toHaveBeenCalledWith('redBlack', 101, false);
    });

    it('should set color gradient for inner circle', () => {
      expect(colorGradient).toHaveBeenCalledWith('yellowBlack', 101, false);
    });
  });

  describe('without known line segment', () => {
    let wrapper;

    beforeAll(() => {
      colorGradient.mockClear();
      wrapper = shallow(
        CircHeatmapLegend({
          known: false,
          circHeatmapSettings: settings,
          segments,
        }),
      );
    });

    it('should to match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should render 3 g elements for gradients', () => {
      expect(wrapper.find('svg > g').length).toBe(3);
    });
  });
});
