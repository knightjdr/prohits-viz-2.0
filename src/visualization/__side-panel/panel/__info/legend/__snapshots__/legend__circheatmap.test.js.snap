// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircHeatmap legend with known line segment should to match snapshot 1`] = `
<svg
  height={270}
  id="legend"
  viewBox="0 0 200 270"
  width="200"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    key="attribute 1"
    transform="translate(0 0)"
  >
    <defs>
      <linearGradient
        id="attribute 1-legendGradient"
      >
        <stop
          offset="0%"
          stopColor="#ffffff"
        />
        <stop
          offset="50%"
          stopColor="#0000ff"
        />
        <stop
          offset="100%"
          stopColor="#000000"
        />
      </linearGradient>
    </defs>
    <g>
      <text
        textAnchor="middle"
        x="100"
        y="20"
      >
        attribute 1
      </text>
      <rect
        fill="url('#attribute 1-legendGradient')"
        height="20"
        width="150"
        x="25"
        y="30"
      />
      <text
        textAnchor="middle"
        x="25"
        y="65"
      >
        0
      </text>
      <text
        textAnchor="middle"
        x="175"
        y="65"
      >
        50
      </text>
    </g>
  </g>
  <g
    key="attribute 1"
    transform="translate(0 70)"
  >
    <defs>
      <linearGradient
        id="attribute 1-legendGradient"
      >
        <stop
          offset="0%"
          stopColor="#ffffff"
        />
        <stop
          offset="50%"
          stopColor="#0000ff"
        />
        <stop
          offset="100%"
          stopColor="#000000"
        />
      </linearGradient>
    </defs>
    <g>
      <text
        textAnchor="middle"
        x="100"
        y="20"
      >
        attribute 1
      </text>
      <rect
        fill="url('#attribute 1-legendGradient')"
        height="20"
        width="150"
        x="25"
        y="30"
      />
      <text
        textAnchor="middle"
        x="25"
        y="65"
      >
        0
      </text>
      <text
        textAnchor="middle"
        x="175"
        y="65"
      >
        50
      </text>
    </g>
  </g>
  <g
    key="attribute 1"
    transform="translate(0 140)"
  >
    <defs>
      <linearGradient
        id="attribute 1-legendGradient"
      >
        <stop
          offset="0%"
          stopColor="#ffffff"
        />
        <stop
          offset="50%"
          stopColor="#0000ff"
        />
        <stop
          offset="100%"
          stopColor="#000000"
        />
      </linearGradient>
    </defs>
    <g>
      <text
        textAnchor="middle"
        x="100"
        y="20"
      >
        attribute 1
      </text>
      <rect
        fill="url('#attribute 1-legendGradient')"
        height="20"
        width="150"
        x="25"
        y="30"
      />
      <text
        textAnchor="middle"
        x="25"
        y="65"
      >
        0
      </text>
      <text
        textAnchor="middle"
        x="175"
        y="65"
      >
        50
      </text>
    </g>
  </g>
  <g
    transform="translate(0 230)"
  >
    <text
      textAnchor="middle"
      x="100"
      y="0"
    >
      Known
    </text>
    <line
      stroke="black"
      strokeWidth="3"
      x1="50"
      x2="150"
      y1="10"
      y2="10"
    />
  </g>
</svg>
`;

exports[`CircHeatmap legend without known line segment should to match snapshot 1`] = `
<svg
  height={270}
  id="legend"
  viewBox="0 0 200 270"
  width="200"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    key="attribute 1"
    transform="translate(0 0)"
  >
    <defs>
      <linearGradient
        id="attribute 1-legendGradient"
      >
        <stop
          offset="0%"
          stopColor="#ffffff"
        />
        <stop
          offset="50%"
          stopColor="#0000ff"
        />
        <stop
          offset="100%"
          stopColor="#000000"
        />
      </linearGradient>
    </defs>
    <g>
      <text
        textAnchor="middle"
        x="100"
        y="20"
      >
        attribute 1
      </text>
      <rect
        fill="url('#attribute 1-legendGradient')"
        height="20"
        width="150"
        x="25"
        y="30"
      />
      <text
        textAnchor="middle"
        x="25"
        y="65"
      >
        0
      </text>
      <text
        textAnchor="middle"
        x="175"
        y="65"
      >
        50
      </text>
    </g>
  </g>
  <g
    key="attribute 1"
    transform="translate(0 70)"
  >
    <defs>
      <linearGradient
        id="attribute 1-legendGradient"
      >
        <stop
          offset="0%"
          stopColor="#ffffff"
        />
        <stop
          offset="50%"
          stopColor="#0000ff"
        />
        <stop
          offset="100%"
          stopColor="#000000"
        />
      </linearGradient>
    </defs>
    <g>
      <text
        textAnchor="middle"
        x="100"
        y="20"
      >
        attribute 1
      </text>
      <rect
        fill="url('#attribute 1-legendGradient')"
        height="20"
        width="150"
        x="25"
        y="30"
      />
      <text
        textAnchor="middle"
        x="25"
        y="65"
      >
        0
      </text>
      <text
        textAnchor="middle"
        x="175"
        y="65"
      >
        50
      </text>
    </g>
  </g>
  <g
    key="attribute 1"
    transform="translate(0 140)"
  >
    <defs>
      <linearGradient
        id="attribute 1-legendGradient"
      >
        <stop
          offset="0%"
          stopColor="#ffffff"
        />
        <stop
          offset="50%"
          stopColor="#0000ff"
        />
        <stop
          offset="100%"
          stopColor="#000000"
        />
      </linearGradient>
    </defs>
    <g>
      <text
        textAnchor="middle"
        x="100"
        y="20"
      >
        attribute 1
      </text>
      <rect
        fill="url('#attribute 1-legendGradient')"
        height="20"
        width="150"
        x="25"
        y="30"
      />
      <text
        textAnchor="middle"
        x="25"
        y="65"
      >
        0
      </text>
      <text
        textAnchor="middle"
        x="175"
        y="65"
      >
        50
      </text>
    </g>
  </g>
</svg>
`;
